src/trace-writer.ts

```diff
@@ -331,10 +333,13 @@ export class TraceWriter extends common.Service {
     const options = {method: 'PATCH', uri, body: json, headers};
     this.logger.info('TraceWriter#publish: Publishing to ' + uri);
     this.request(options, (err, body?, response?) => {
-      const statusCode = (response && response.statusCode) || 'unknown';
+      const statusCode = response && response.statusCode;
       if (err) {
-        this.logger.error(`TraceWriter#publish: Received error status code ${
-            statusCode}. Original error: ${err}`);
+        this.logger.error(`TraceWriter#publish: Received error ${
+            statusCode ?
+                `with status code ${statusCode}` :
+                ''} while publishing traces to cloudtrace.googleapis.com: ${
```

Comment:
nit: would it make sense to use ${uri} here instead of duplicating the address? Or maybe extract a hostname variable.